src/farneback.jl

```diff
@@ -0,0 +1,193 @@
+
+                sub_A1 = A1[i,j,:,:]
+                sub_B1 = B1[i,j,:]
+                sub_A2 = A2[mod_i,mod_j,:,:]
+                sub_B2 = B2[mod_i,mod_j,:]
+
+                A = (sub_A1 .+ sub_A2)./2
+                dB = A*d .- ((sub_B2 .- sub_B1)./2)
+
+                AtA[i,j,:,:] = A'*A
+                AtB[i,j,:] = A'*dB
+            end
+        end
+
+        len = floor(Int, algo.window_size/2)
+        w = ImageFiltering.KernelFactors.gaussian((algo.σw),(algo.window_size))[-len:len]
+
+        for i = 1:2
+            for j = 1:2
+                if algo.gauss_flag
+                    AtA[:,:,i,j] = conv2(w, w, AtA[:,:,i,j])[1+len:size(first_img)[1]+len,1+len:size(first_img)[2]+len]
+                else
+                    AtA[:,:,i,j] = conv2(ones(algo.window_size)./algo.window_size, ones(algo.window_size)./algo.window_size, AtA[:,:,i,j])[1+len:size(first_img)[1]+len,1+len:size(first_img)[2]+len]
+                end
+            end
+            if algo.gauss_flag
+                AtB[:,:,i] = conv2(w, w, AtB[:,:,i])[1+len:size(first_img)[1]+len,1+len:size(first_img)[2]+len]
+            else
+                AtB[:,:,i,j] = conv2(ones(algo.window_size)./algo.window_size, ones(algo.window_size)./algo.window_size, AtB[:,:,i,j])[1+len:size(first_img)[1]+len,1+len:size(first_img)[2]+len]
+            end
+        end
+
+        for i = 1:size(first_img)[1]
+            for j = 1:size(first_img)[2]
+                disp[i,j][:] = MVector{2,Float64}(pinv(AtA[i,j,:,:])*AtB[i,j,:])
+            end
+        end
+    end
+
+    return disp
+end
+
+"""
+```
+A, B, C = polynomial_expansion(img, neighbourhood, σ)
+```
+
+Returns the polynomial coefficients of the approximation of the neighbourhood of a
+point in a 2D function with a polynomial.
+The expansion is: f(x) = x'Ax + B'x + C
+
+Parameters:
+
+-  img               =  Grayscale image who polynomial expansion is to be found
+-  neighbourhood     =  Size of the pixel neighbourhood used to find polynomial expansion for each pixel;
+                        larger values mean that the image will be approximated with smoother surfaces,
+                        yielding more robust algorithm and more blurred motion field
+-  σ                 =  Standard deviation of the Gaussian that is used to smooth derivatives used as a
+                        basis for the polynomial expansion (Applicability)
+
+## References
+
+Farnebäck, G.: Polynomial Expansion for Orientation and Motion Estimation. PhD thesis, Linköping University,
+Sweden, SE-581 83 Linköping, Sweden (2002) Dissertation No 790, ISBN 91-7373-475-6.
+
+"""
+
+function polynomial_expansion(img::AbstractArray{T, 2}, neighbourhood::Int, σ::Real) where T <: Gray
+    len = floor(Int, neighbourhood/2)
+    a = ImageFiltering.Kernel.gaussian((σ,σ), (neighbourhood,neighbourhood))[-len:len, -len:len]
+
+    c = ones(size(img))
+    padded_c = reshape(padarray(c, Fill(0, (len, len)))[:], size(img) .+ 2*len)
```

Comment:
If you want to work with a canonically indexed array instead of an `OffsetArray` then do this instead:
`padded_c = parent(padarray(c, Fill(0, (len, len))))`
`padded_img = parent(padarray(img, Fill(0, (len, len))))`

Documentation for `parent`. 
```julia
  parent(A)

  Returns the "parent array" of an array view type (e.g., SubArray), or the array itself if it is
  not a view.
```


Your current approach of vectorization and reshaping  allocates unnecessary memory.